### PR TITLE
Hearing - New XEH, Fix dead effects, Medical var

### PR DESCRIPTION
**Pull request by PabstMirror**
**When merged this pull request will:**
- CBA/addClassEventHandlers optimization
- Fix ear-ringing playing for units in spectator
- Medical now uses hearing's `unconsciousnessVolume` if it exists

### CODE REVIEW

**Comment by esteldunedain** [comment]
@PabstMirror: As far as I can tell `ace_hearing_fnc_updateVolume` already tweaks the volume for unconscious units. I really don't see the point of medical doing the same; anyway, this PR should remove that tweak if that's what we want.

**Comment by Glowbal** [comment]
In #3765 I linked this function: https://github.com/acemod/ACE3/blob/master/addons/common/functions/fnc_setHearingCapability.sqf

A solution like this would be ideal imo. It would not only support us modifying hearing volume, but also give support for third party mods doing so, without conflicting with anything we do.

**Comment by PabstMirror** [comment]
Update to use function - Will close #3765

**Comment by PabstMirror** [comment]
@esteldunedain - I agree, I removed setting the global volume in medical.
It will still set the outgoing voice settings on unconscious.

File: addons/hearing/functions/fnc_updateVolume.sqf [hunk 1]
```diff
@@ -16,6 +16,12 @@
  */
 #include "script_component.hpp"
 
+if (!alive ACE_player) then {
```
Comment (Glowbal) [inline addons/hearing/functions/fnc_updateVolume.sqf]: Can this not be `exitWith` ? If `ACE_player` is not alive, is anything that comes after it still relevant?

**Pull request merged**